app/src/pages/inside/launchesPage/modals/launchMergeModal/launchMergeModal.jsx

```diff
@@ -0,0 +1,304 @@
+import { injectIntl, defineMessages, intlShape } from 'react-intl';
+import { ModalLayout, withModal, ModalField, ModalContentHeading } from 'components/main/modal';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+import { FieldProvider } from 'components/fields/fieldProvider';
+import { Input } from 'components/inputs/input';
+import { InputTextArea } from 'components/inputs/inputTextArea';
+import { InputRadio } from 'components/inputs/inputRadio';
+import { InputTagsSearch } from 'components/inputs/inputTagsSearch';
+import { InputCheckbox } from 'components/inputs/inputCheckbox';
+import { activeProjectSelector, userInfoSelector } from 'controllers/user';
+import { MergeTypeScheme } from './mergeTypeScheme';
+import { StartEndTime } from './startEndTime';
+import styles from './launchMergeModal.scss';
+
+const MERGE_TYPE_DEEP = 'DEEP';
+const MERGE_TYPE_BASIC = 'BASIC';
+const FIELD_LABEL_WIDTH = 130;
+const DESCRIPTION_SEPARATOR = '\n≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡\n';
+const cx = classNames.bind(styles);
+const messages = defineMessages({
+  MergeLaunchHeader: {
+    id: 'MergeLaunchDialog.MergeLaunchHeader',
+    defaultMessage: 'Merge launches',
+  },
+  mergeTypeHeading: {
+    id: 'MergeLaunchDialog.mergeTypeHeading',
+    defaultMessage: 'Choose type of merge',
+  },
+  launchInfoHeading: {
+    id: 'MergeLaunchDialog.launchInfoHeading',
+    defaultMessage: 'All selected launches will be merged in one',
+  },
+  mergeTypeLinear: {
+    id: 'MergeLaunchDialog.mergeTypeLinear',
+    defaultMessage: 'Linear merge',
+  },
+  mergeTypeDeep: {
+    id: 'MergeLaunchDialog.mergeTypeDeep',
+    defaultMessage: 'Deep merge',
+  },
+  launchNameLabel: {
+    id: 'MergeLaunchDialog.launchNameLabel',
+    defaultMessage: 'Launch name',
+  },
+  launchNamePlaceholder: {
+    id: 'MergeLaunchDialog.launchNamePlaceholder',
+    defaultMessage: 'Enter launch name',
+  },
+  launchOwnerLabel: {
+    id: 'MergeLaunchDialog.launchOwnerLabel',
+    defaultMessage: 'Owner',
+  },
+  launchDescriptionLabel: {
+    id: 'MergeLaunchDialog.launchDescriptionLabel',
+    defaultMessage: 'Description',
+  },
+  launchTagsLabel: {
+    id: 'MergeLaunchDialog.launchTagsLabel',
+    defaultMessage: 'Tags',
+  },
+  launchTagsPlaceholder: {
+    id: 'MergeLaunchDialog.launchTagsPlaceholder',
+    defaultMessage: 'Enter tag name',
+  },
+  launchTagsHint: {
+    id: 'MergeLaunchDialog.launchTagsHint',
+    defaultMessage: 'Please enter 1 or more characters',
+  },
+  launchTimeLabel: {
+    id: 'MergeLaunchDialog.launchTimeLabel',
+    defaultMessage: 'Time start/end',
+  },
+  extendSuitesDescriptionText: {
+    id: 'MergeLaunchDialog.extendSuitesDescriptionText',
+    defaultMessage:
+      "Extend child suites description with original launch name (e.g. @launch 'Launch #1')",
+  },
+});
+
+@withModal('launchMergeModal')
+@injectIntl
+@reduxForm({
+  form: 'launchMergeForm',
```

Comment:
form name is used in many places, so it would be more convenient to define it as a constant